webapp/src/main/java/com/google/sps/servlets/UploadServlet.java

```diff
@@ -76,13 +83,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             // Save the file on disk
             item.write(storeFile);
 
-            // TO DO: Call validator here 
+            // TO DO: get feed name and thread specification from form
+            String validatorOutputJson = runValidator(filePath, 1, "nl-openov");
+
             response.getWriter().println("Upload has been done successfully!");
+            // Print the json output 
+            response.getWriter().println(validatorOutputJson);
           }
         }
       }
     } catch (Exception ex) {
       response.getWriter().println("There was an error: " + ex.getMessage());
     }
   }
+
+  // Calls the global mobility validator function to load and validate the transit data
+  private String runValidator(String filePath, int numThreads, String feedNameString) {
+    ValidatorLoader validatorLoader = new ValidatorLoader();
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+
+    GtfsFeedName feedName = GtfsFeedName.parseString(feedNameString);
+    final long startNanos = System.nanoTime();
+
+    // Input.
+    feedLoader.setNumThreads(numThreads);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsFeedContainer feedContainer;
+    GtfsInput gtfsInput;
+
+    try {
+      gtfsInput = GtfsInput.createFromPath(Paths.get(filePath));
```

Comment:
Can this be tested?

Reply:
I will figure out how to write the tests now

Reply:
Thanks

Reply:
I just created 2 simple tests that ensure it reads file and generates the correct json